File: CourierApp.Shared/DTOs/VehicleDto.cs
using System.ComponentModel.DataAnnotations;

namespace CourierApp.Shared.DTOs;

public class VehicleDto
{
    [Required]
    public string VehicleId { get; set; } = string.Empty;
    
    [Required]
    public string Brand { get; set; } = string.Empty;
    
    [Required]
    public string Model { get; set; } = string.Empty;
    
    [Required]
    public string LicensePlate { get; set; } = string.Empty;
    
    [Required]
    public int PackageCapacity { get; set; }
}

File: courier-app.server/src/vehicle/dto/vehicle.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsString, IsInt } from 'class-validator';

export class VehicleDto {
  @ApiProperty()
  @IsString()
  vehicleId: string;

  @ApiProperty()
  @IsString()
  brand: string;

  @ApiProperty()
  @IsString()
  model: string;

  @ApiProperty()
  @IsString()
  licensePlate: string;

  @ApiProperty()
  @IsInt()
  packageCapacity: number;
}